From 15f9d11019eb3c3499fb3f9cd3c07a3733d4041f Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Tue, 21 Nov 2017 17:03:28 +0800
Subject: [PATCH 2851/5242] MLK-16924: hdp: Disable HDP hotplug detect thread
 temporary

commit  d3c2cafc1bce78e0b9c5779f17dccc3599f7febf from
https://source.codeaurora.org/external/imx/linux-imx.git

HDMI Fw is unstable, hotplug detect thread may
blocked at hdmi mailbox access.
Disable hotplug detect thread function temporary.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 2bddeba..680bffa 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -622,6 +622,7 @@ static void imx_hdp_mode_setup(struct imx_hdp *hdp, struct drm_display_mode *mod
 	hdp->vic = drm_match_cea_mode(mode);
 }
 
+#if 0
 static int imx_hdp_cable_plugin(struct imx_hdp *hdp)
 {
 	return 0;
@@ -632,7 +633,7 @@ static int imx_hdp_cable_plugout(struct imx_hdp *hdp)
 	imx_hdp_call(hdp, pixel_clock_disable, &hdp->clks);
 	return 0;
 }
-
+#endif
 
 static void imx_hdp_bridge_mode_set(struct drm_bridge *bridge,
 				    struct drm_display_mode *orig_mode,
@@ -962,6 +963,7 @@ static int imx8qm_hdp_swrite(struct hdp_mem *mem, unsigned int addr, unsigned in
 };
 MODULE_DEVICE_TABLE(of, imx_hdp_dt_ids);
 
+#if 0
 #ifdef hdp_irq
 static irqreturn_t imx_hdp_irq_handler(int irq, void *data)
 {
@@ -1033,6 +1035,7 @@ static int hpd_det_worker(void *_dp)
 	return 0;
 }
 #endif
+#endif
 
 static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 			    void *data)
@@ -1047,7 +1050,9 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 	struct drm_bridge *bridge;
 	struct drm_connector *connector;
 	struct resource *res;
+#if 0
 	struct task_struct *hpd_worker;
+#endif
 	int irq;
 	int ret;
 
@@ -1174,6 +1179,7 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 		return ret;
 	}
 
+#if 0
 #ifdef hdp_irq
 	ret = devm_request_threaded_irq(dev, irq,
 					NULL, imx_hdp_irq_handler,
@@ -1190,6 +1196,7 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 
 	wake_up_process(hpd_worker);	/* avoid contributing to loadavg */
 #endif
+#endif
 
 	return 0;
 #ifdef hdp_irq
-- 
1.7.9.5

